src/main/java/org/broadinstitute/hellbender/utils/locusiterator/AlignmentContextIteratorBuilder.java

```diff
@@ -0,0 +1,178 @@
+package org.broadinstitute.hellbender.utils.locusiterator;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SAMSequenceDictionary;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.engine.AlignmentContext;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.iterators.IntervalLocusIterator;
+import org.broadinstitute.hellbender.utils.iterators.IntervalOverlappingIterator;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Create an iterator for traversing alignment contexts in a specified manner.  This class should be able to support both spark and non-spark LocusWalker
+ */
+public class AlignmentContextIteratorBuilder {
+
+    protected static final Logger logger = LogManager.getLogger(AlignmentContextIteratorBuilder.class);
+
+    private boolean isEmitEmptyLoci;
+    private boolean isKeepUniqueReadListInLibs;
+    private boolean isIncludeDeletions;
+    private boolean isIncludeNs;
+    private LIBSDownsamplingInfo downsamplingInfo;
+
+    public boolean isEmitEmptyLoci() {
+        return isEmitEmptyLoci;
+    }
+
+    public void setEmitEmptyLoci(boolean emitEmptyLoci) {
```

Comment:
I also expect setters to return void, but I thought that the builder pattern is usually returning `this` as a convention. Anyway, it was a suggestion to use the builder with chaining and not making a variable to call the setters in...